src/test/java/org/junit/tests/assertion/AssertionTest.java

```diff
@@ -83,11 +98,29 @@ public void arraysActualNullMessage() {
     }
 
     @Test
-    public void arraysDifferentLengthMessage() {
+    public void arraysDifferentLengthDifferingAtStartMessage() {
+        try {
+            assertArrayEquals("not equal", (new Object[]{true}), (new Object[]{false, true}));
+        } catch (AssertionError exception) {
+            assertEquals("not equal: array lengths differed, expected.length=1 actual.length=2; arrays first differed at element [0]; expected:<true> but was:<false>", exception.getMessage());
+        }
```

Comment:
Before this line add a `return` statement, then after the `catch` block call `fail()` so we know an exception was thrown.

This will be so much easier if we could use lamdas and `Assert.expectThrows()` ...